src/app/config/resolve_addr.c

```diff
@@ -481,51 +527,24 @@ find_my_address(const or_options_t *options, int family, int warn_severity,
    * Step 1: Discover address by attempting 3 different methods consecutively.
    */
 
-  /* Attempt #1: Get address from configuration. */
-  ret = get_address_from_config(options, warn_severity, family, &method_used,
-                                &hostname_used, &my_addr);
-  if (ret == 0) {
-    log_fn(warn_severity, LD_CONFIG, "Address found in configuration: %s",
-           fmt_addr(&my_addr));
-  } else {
-    /* Unable to resolve an Address statement is a failure. Also, using
-     * default dirauth error means that the configured address is internal
-     * which is only accepted if custom authorities are used. */
-    if (ret == ERR_FAIL_RESOLVE || ret == ERR_DEFAULT_DIRAUTH) {
+  /* Go over the function table. They are in order. */
+  for (size_t idx = 0; idx < fn_address_table_len; idx++) {
+    fn_address_ret_t ret = fn_address_table[idx](options, warn_severity,
+                                                 family, &method_used,
+                                                 &hostname_used, &my_addr);
+    if (ret == FN_RET_BAIL) {
       return false;
+    } else if (ret == FN_RET_OK) {
+      goto found;
```

Comment:
We used to log where the address was found.  I think we should keep doing that, or the debugging will be really hard.

Reply:
Yes I moved all logging into the function itself. If you look in all `get_address_*` function, when it returns `FN_RET_OK`, it logs that we found the address.

Good enough?